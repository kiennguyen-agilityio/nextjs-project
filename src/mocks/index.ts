export const categoryOptions = [
  { value: 'option1', label: 'Option 1' },
  { value: 'option2', label: 'Option 2' },
  { value: 'option3', label: 'Option 3' },
];

export const listCategory = [
  {
    id: '1',
    label: 'Email',
    options: [
      { value: 'email1', label: 'email1@example.com' },
      { value: 'email2', label: 'email2@example.com' },
    ],
  },
  {
    id: '2',
    label: 'Full Name',
    options: [
      { value: 'john_doe', label: 'John Doe' },
      { value: 'jane_doe', label: 'Jane Doe' },
    ],
  },
  {
    id: '3',
    label: 'Role',
  },
  {
    id: '4',
    label: 'Joined',
    options: [
      { value: '2021', label: '2021' },
      { value: '2022', label: '2022' },
    ],
  },
];
