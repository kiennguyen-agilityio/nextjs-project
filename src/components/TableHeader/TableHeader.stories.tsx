import type { Meta, StoryObj } from '@storybook/react';

// components
import TableHeader from '@/components/TableHeader';

// mocks
import { roles } from '@/mocks';

const meta = {
  title: 'Components/TableHeader',
  component: TableHeader,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof TableHeader>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    roles: roles,
  },
};
