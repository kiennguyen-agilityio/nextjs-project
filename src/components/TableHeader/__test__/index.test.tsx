import { render } from '@testing-library/react';

// components
import TableHeader from '@/components/TableHeader';

// mocks
import { roles } from '@/mocks';

jest.mock('next/navigation', () => ({
  useRouter() {
    return {
      push: () => jest.fn(),
      replace: () => jest.fn(),
    };
  },
  usePathname() {
    return '';
  },
  useSearchParams() {
    return new URLSearchParams();
  },
}));
test('renders Category component with dropdown buttons', () => {
  const { asFragment } = render(
    <table>
      <TableHeader roles={roles} />
    </table>,
  );
  expect(asFragment()).toMatchSnapshot();
});
