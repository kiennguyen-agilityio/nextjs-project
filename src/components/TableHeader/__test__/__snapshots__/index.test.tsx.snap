// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Category component with dropdown buttons 1`] = `
<DocumentFragment>
  <table>
    <thead
      class="bg-gray-400 text-gray-800 border-b border-gray-300"
    >
      <tr
        class="flex items-center justify-between pl-6 pr-8 py-3"
      >
        <th
          class="text-left font-semibold text-sm sm:text-base px-4 py-2"
        >
          <select
            class="inline-flex justify-center rounded-md w-full px-2 py-1 text-sm font-medium text-gray-700 cursor-pointer focus:outline-none max-w-40 undefined"
            name="userRole"
          >
            <option
              value="1"
            >
              Email
            </option>
          </select>
        </th>
        <th
          class="text-left font-semibold text-sm sm:text-base px-4 py-2"
        >
          <select
            class="inline-flex justify-center rounded-md w-full px-2 py-1 text-sm font-medium text-gray-700 cursor-pointer focus:outline-none max-w-40 undefined"
            name="userRole"
          >
            <option
              value="2"
            >
              Full Name
            </option>
            <option
              value="asc"
            >
              Full Name: Asc
            </option>
            <option
              value="desc"
            >
              Full Name: Desc
            </option>
          </select>
        </th>
        <th
          class="text-left font-semibold text-sm sm:text-base px-4 py-2"
        >
          <select
            class="inline-flex justify-center rounded-md w-full px-2 py-1 text-sm font-medium text-gray-700 cursor-pointer focus:outline-none max-w-40 undefined"
            name="userRole"
          >
            <option
              value="3"
            >
              Role
            </option>
            <option
              value="1"
            >
              Admin
            </option>
            <option
              value="2"
            >
              User
            </option>
            <option
              value="3"
            >
              Guest
            </option>
          </select>
        </th>
        <th
          class="text-left font-semibold text-sm sm:text-base px-4 py-2"
        >
          <select
            class="inline-flex justify-center rounded-md w-full px-2 py-1 text-sm font-medium text-gray-700 cursor-pointer focus:outline-none max-w-40 undefined"
            name="userRole"
          >
            <option
              value="4"
            >
              Joined
            </option>
            <option
              value="asc"
            >
              Joined: Asc
            </option>
            <option
              value="desc"
            >
              Joined: Desc
            </option>
          </select>
        </th>
        <th
          class="text-left px-4 py-2"
        />
      </tr>
    </thead>
  </table>
</DocumentFragment>
`;
